client/src/components/application/Form.tsx

```diff
@@ -73,8 +74,18 @@ export default function Form({
   const handleSubmitApplication = async () => {
     setSubmitted(true);
     await validate();
-    if (formValidation.valid) {
-      dispatch(submitApplication(round.address, formInputs));
+    if (roundManagerClient && formValidation.valid) {
+      showToast(true);
+      await submitApplication(roundManagerClient, round.address, formInputs)
```

Comment:
Maybe stick to either `await` or `.then()` here?

Reply:
Picking this up